puts/harness/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(puts VERSION 0.1 LANGUAGES C)
+
+enable_testing()
+
+set(PUTS "")
+set(PUT_LIBS "")
+
+string(REPLACE "," ";" VENDORS "${VENDORS}")
+foreach(vendor_path IN LISTS VENDORS)
+    set(PROTOCOL "tls")
+    set(HARNESS "openssl")
```

Comment:
We need to document these parameters somewhere. Not sure what a good way would be

Reply:
This will come in a future PR. The idea will be that each C PUT harness advertises through cmake which protocol it supports and which vendor library it can wrap (OpenSSL, LibreSSL, ... maybe with version ranges). Then when we get the VENDORS list here, we check which vendor library version it is and select the most appropriate harness. So these variables will not stay hardcoded like this.